templates/periodicPayment.go

```diff
@@ -0,0 +1,99 @@
+package templates
+
+import (
+	"encoding/base64"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// PeriodicPayment template representation
+type PeriodicPayment struct {
+	ContractTemplate
+	amount      uint64
+	receiver    types.Address
+	leaseBase64 string
+}
+
+// GetWithdrawalTransaction returns a signed transaction extracting funds from the contract
+// fee: the fee to pay in microAlgos
+// firstValid: the first round on which the txn will be valid
+// lastValid: the final round on which the txn will be valid
+// genesisHash: the hash representing the network for the txn
+func (c PeriodicPayment) GetWithdrawalTransaction(fee, firstValid, lastValid uint64, genesisHash []byte) ([]byte, error) {
```

Comment:
```
&&
txn FirstValid
int TMPL_PERIOD
%
int 0
==
&&
txn LastValid
int TMPL_DUR
txn FirstValid
+
==
```


This is from the contract. `FirstValid` must be a multiple of `period` and `LastValid` must be equal to `duration + firstValid` so worth checking that and computing `LastValid` ourselves